Migrate POST and GET operation

RamOwen99 opened the pull request:
# Migrate the POST operation to create a new user and GET operation to get all users

Watch more information in [trello](https://trello.com/c/JHGtpUpU/104-us-52-imp-check-the-current-post-operation-and-get-operation-of-at-sso-api)

## Unit testing in the controller
![imagen](https://user-images.githubusercontent.com/78741607/115874326-95cf8c00-a409-11eb-9a45-7a8e17e1d066.png)

## Unit testing in the service
![imagen](https://user-images.githubusercontent.com/78741607/115874375-a7189880-a409-11eb-9898-d2f204d19228.png)

## Unit testing in the validations when we need to throw an exception
![imagen](https://user-images.githubusercontent.com/78741607/115874439-b697e180-a409-11eb-81c2-3f1baf8c6ca1.png)

RamOwen99 committed:
Move ExceptionMessage in other package

I move the constants messages in a new package in the exceptions package
to identify more quick where are all the messages when an exception is
throw it

RamOwen99 committed:
Create validations in a file

I delete in the service package the BusinessValidations class to
refactor the validations to create a user in a package called validator.
In this package will be all the validations that we required.

graciela24 committed:
Add OrikaMapper dependency

graciela24 committed:
Add class UserMapping

graciela24 committed:
Add Update Request and Response

RamOwen99 committed:
Modify the badRequestExceptions and messages

Create an enum to store all the constant messages that we will throw
when an exception happens

graciela24 committed:
Update clases' properties

graciela24 committed:
Add class New User Request and Response

RamOwen99 committed:
Update pom.xml

Add the Junit version, exclude some package from the Jacoco coverage

RamOwen99 committed:
Delete unnecessary files

RamOwen99 committed:
Import static constant

RamOwen99 committed:
Use validation from UserValidator and Log4j2

RamOwen99 committed:
User Json layout template

RamOwen99 committed:
Rename ControllerApplication

I rename this clase because the name has a grammatical mistake.
Also I update some structure in the post and get methods

RamOwen99 committed:
Overloading construction method

RamOwen99 committed:
Refactore service

I change the structure of the service to use orikaMapper, create a
method to set the first and last name in capital letters, move the
validation whether the user's email is already exist in the DB

RamOwen99 committed:
Move method to validate an existing email

RamOwen99 committed:
Create unit testing

The unit testing belongs to the validations and the service

RamOwen99 committed:
Create unit testing

The unit testing belongs to the validations and the service

RamOwen99 committed:
Merge branch 'feature/US_052_imp_current_post_and_get_operation' of https://github.com/at-internship/at-sso-api into feature/US_052_imp_current_post_and_get_operation

RamOwen99 committed:
Delete unnecessary docs

RamOwen99 committed:
Update dependencies

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/controller/GlobalExceptionHandler.java:
```diff
@@ -1,44 +0,0 @@
-package com.agilethought.internship.sso.controller;
```
You should not delete this class, because belongs to previous implementation and we want preserve it

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/exception/BadRequestException.java:
```diff
@@ -3,25 +3,11 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
-import lombok.Getter;
```
we need to follow same logic that was implemented.

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/exception/HttpMessageNotReadableException.java:
```diff
@@ -1,27 +0,0 @@
-/*package com.agilethought.internship.sso.exception;
```
Don't remove it

RamOwen99 replied:
I delete this file (`exception/HttpMessageNotReadableException.java `) because it was all commented, and the `errorhandling/` files I decided to made an enum

ivanflores15 replied:
don't remove. in near future we will review what the purpose is for this class

ivanflores15 commented on src/main/java/com/agilethougth/intership/sso/errorhandling/HttpExceptionMessage.java:
```diff
@@ -1,17 +0,0 @@
-package com.agilethougth.intership.sso.errorhandling;
```
maybe this class will be required with rollback of exception handler

ivanflores15 commented on src/main/java/com/agilethougth/intership/sso/errorhandling/PathErrorMessage.java:
```diff
@@ -1,6 +0,0 @@
-package com.agilethougth.intership.sso.errorhandling;
```
maybe this class will be required with rollback of exception handler

ivanflores15 commented on src/main/resources/log4j2.properties:
```diff
@@ -1,29 +1,20 @@
+status=error
```
Go back to previous implementation. It is not part of your US

RamOwen99 committed:
Return previous implementations

ivanflores15 commented:
please rebase to current develop reference

RamOwen99 committed:
Merge branch 'develop' into feature/US_052_imp_current_post_and_get_operation

ivanflores15 commented on .gitignore:
```diff
@@ -6,6 +6,10 @@ target/
 bin/
 sprintsLogs/
 
+.mvn/
```
repeated lines inside, lines 6, 7...

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/controller/ControllerApplication.java:
```diff
@@ -15,21 +17,21 @@
 	@Autowired
 	ServiceApplication serviceApplication;
 
 	@PostMapping(value = "/api/v1/user", produces = "application/json")
```
use `@RequestMapping` to define base route api path

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/controller/ControllerApplication.java:
```diff
@@ -15,21 +17,21 @@
+		return serviceApplication.createUser(request);
+	}
 
 	@GetMapping(value = "/api/v1/user", produces = "application/json")
```
`"/api/v1/user"` resources must be in plural "users"

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/controller/GlobalExceptionHandler.java:
```diff
@@ -13,32 +12,36 @@
+
+    @ExceptionHandler
+    public final ResponseEntity<ExceptionResponse> handleBadRequestException(BadRequestException e) {
+        ExceptionResponse exceptionResponse = new ExceptionResponse();
```
When is setting ``` date, new Date(), e.getStatus(), HttpStatus.BAD_REQUEST, e.getMessage(), e.getPath()``` ? because we will have a response entity with null values

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/domain/UserDTO.java:
```diff
@@ -21,9 +22,12 @@
 	private Integer status;
 
-	public UserDTO(String name, String firstName, String lastName, String email, String password, Integer status) {
+	public UserDTO() {
```
Why not use lombok annotations?

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/exception/BadRequestException.java:
```diff
@@ -3,25 +3,11 @@
-		setMessage(message);
-		setPath(path);		
-		setStatus(r.value());
+	public BadRequestException(String message){
```
Do not change this class, exceute a rollback of this class

ivanflores15 commented on src/main/java/com/agilethought/internship/sso/exception/ExceptionResponse.java:
```diff
@@ -16,8 +16,11 @@
-	
-	public ExceptionResponse(Date timestamp,int status,HttpStatus error, String message, String path) {
+
+	public ExceptionResponse() {
```
use lombok annotations, unless you have a good reason to not use it

RamOwen99 committed:
Delete duplicates lines

RamOwen99 committed:
Put base route and use a constructor in exception

RamOwen99 committed:
Update classes

In the UserDTO, ExceptionResponse use Lombok.
In the ErrorMessageCreateUser, I add a new constant message

RamOwen99 committed:
Update the bad request constructor

RamOwen99 committed:
Update the URL request
